### ISSUE TITLE
Incorrect JSON generated for a parameter example

### ISSUE THREAD

**Issue by Leon99** [mattfrear/Swashbuckle.AspNetCore.Filters#69]
We're getting warnings from ReDoc (e.g. _Other properties are defined at the same level as $ref at "#/paths/~1quotes/post/parameters/0/schema". They are IGNORED according to the JsonSchema spec_). Upon investigation, it looks like `example` for a parameter is added under `schema`, instead of next to `schema`, as per the specs (https://swagger.io/docs/specification/adding-examples/). Example that causes the warning:
```
    "/quotes": {
      "post": {
        "tags": ["Quotes"],
        "summary": "Create a new quote",
        "operationId": "CreateQuote",
        "consumes": ["application/json"],
        "produces": ["application/json"],
        "parameters": [{
          "name": "data",
          "in": "body",
          "required": false,
          "schema": {
            "$ref": "#/definitions/QuotingRequestValidated",
            "example": {
              "payment_type": "Provider",
              "country_code": "AU",
              "payment_destinations": [{
                  "payee_id": 10759,
                  "payer_amount": 10000.0
                },
                {
                  "payee_id": 163,
                  "payer_amount": 5000.0
                }
              ]
            }
          }
        }],
       "responses": {
       ...
```

**Comment by mattfrear** [comment]
The version of the Swagger spec which you have linked to is Swagger aka OpenDoc v3.0. 

Swashbuckle supports only Swagger 2.0, which has a different and slightly ambiguous section around examples.

Request examples spec (which I quote in my ReadMe):
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#schema-object

Response examples spec:
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#response-object

What I am producing is valid Swagger 2.0, so I probably won't change this.
That said, I am open to someone else making changes to fit the Swagger 3.0 spec, so long as the request and response examples still work in Swagger 2.0, if that is possible. You are welcome to try. Personally, I don't have time to try.

**Comment by Leon99** [comment]
Yeah, I couldn't find any clear definition of how it should be handled in 2.0. However, upon further investigation, I realized that the example is getting included twice - once under `schema` (always) and then again under the definition (if `schema` references it). According to a comment in RequestExample.cs, 
> // now that we have the name, additionally set the example on the object in the schema registry (this is what swagger-ui will display)

which is followed by including the example under the definition.

So I really don't understand why include the example under `schema` if it references a definition, even though the 2.0 specs may not be clear about it. When I think about it, the 3.0 specs that mentions that everything next to `$ref` should be ignored makes total sense - it should go under the referenced definition.
Changing it fixes the issue - if you're ok with it, I'll submit a PR, it's really about moving one line of code inside a conditional (and adjusting two unit tests).

**Comment by mattfrear** [comment]
I suspect that based on the comment "(this is what swagger-ui will display)", that if you remove the example from the schema then it will stop being displayed in swagger-ui. 

You are welcome to try... If you can run the included test project and still see the request examples in swagger-ui then I will accept your PR. 

FWIW, most of my users only care about seeing the examples in swagger-ui, they don't care whether the underlying swagger is valid or not. They often don't even know that it exists.
